ecosystem/sep-0006.md

```diff
@@ -1010,6 +1011,22 @@ Name | Type | Description
 * `too_large` -- deposit/withdrawal size exceeded `max_amount`.
 * `error` -- catch-all for any error not enumerated above.
 
+### Refunds Object Schema
+
+Name | Type | Description
+-----|------|------------
+`type` | string | One of `partial` or `full`
+`amount_refunded` | string | The total amount refunded to the user, in units of `amount_in_asset`. If `type` is `full`, this amount must match `amount_in`.
+`payments` | array | A list of objects containing information on the individual payments made back to the user. The schema for these objects is defined in the section below.
+
+#### Refund Transactions Object Schema
+
+Name | Type | Description
+-----|------|------------
+`id` | string | The payment ID that can be used to identify the refund payment. This is either a Stellar transaction hash or an off-chain payment identifier, such as a reference number provided to the user when the refund was initiated.
+`id_type` | string | `stellar` or `external`.
+`amount` | string | The amount sent back to the user for the payment identified by `id`.
+
```

Comment:
It's common for exchanges to charge a refund fee in case something goes wrong with the transaction. Shouldn't we include some kind of `fee` field as well?

Reply:
For Stellar network payments, the anchor could add 2 transactions, one for the refund and the other for the refund fee, but that may not be practical for bank transfers.

Reply:
Sure, I can add a `fee` attribute. Shouldn't we have a `total_fee` attribute or something similar alongside `amount_refunded` then?

Reply:
Is the idea with the fee attribute that this would represented an amount that the user no longer has a right to, but was not included in the refund transaction? How would this affect the amount refunded?

For example:
- Original deposit $100.
- Anchor needs to refund $25, charges $5 per refund.
- Amount refunded = $25.
- Refunds includes a single refund with id and amount.
- What value does the `refunds.payments[0].amount` field hold? Is it $25, or $20?

Reply:
My understanding is that it would look like this:

* Original deposit $100
* Anchor needs to refund $25, and charges $5 per refund

So:
```json
{
  ...
  "amount_in": "100",
  "amount_out": "65",
  "amount_fee": "5",
  "refunded": false,
  "refunds": {
    "amount_refunded": "25",
    "total_fee": "5",
    "payments": [
      {
        "id": "<stellar hash>"
        "id_type": "stellar",
        "amount": "25",
        "fee": "5"
      }
    ]
  }
}
```

Reply:
That makes sense to me too, since it's consistent with the amount_in description:



`amount_in` | string | (optional) Amount received by anchor at start of transaction as a string with up to 7 decimals. Excludes any fees charged before the anchor received the funds.
-- | -- | --

Reply:
Got it, so is it correct that this formula should balance to zero?

```
amount_in
– amount_out
– amount_fee
+ refunds.amount_refunded
– refunds.total_fee (this should be named amount_fee I think)
```

And:
```
refunds.amount_refunded = sum(refunds.payments[].amount
refunds.total_fee = sum(refunds.payments[].fee)
```

Reply:
Yes thats right. In fact I think those formulas are worth including in the specifications.

Reply:
>Amount received by anchor at start of transaction as a string with up to 7 decimals. Excludes any fees charged before the anchor received the funds.

The excludes statement is ambiguous to me. When I read it I can't tell if it means the amount_in has already had fees excluded, i.e. I'm depositing $100, but amount_in will be $95 because it excludes the fee I paid.